packages/embed-widget/src/App.tsx

```diff
@@ -60,6 +88,43 @@ function App(): JSX.Element {
             throw new Error('Missing URL parameter "name"');
           }
 
+          const sessionDetails = await getSessionDetails();
```

Comment:
There seems to be a lot of duplicate code here as `AppInit` `initApp` effect. Any chance we could split out the common code into a `useInitApp` hook or something?

Reply:
Probably. I thought about it and initially had most of the new code in a separate effect, but it actually caused a race condition with loading the table. I think I'd name it `useInitRedux` or something like that.

I did also try moving some of the redux setters to their respective `*Bootstrap` components. But would need to hoist the redux store up since right now it's a child of all the bootstraps. Shouldn't have an impact though if that's what we went with

Reply:
Instead of hook could just be an async helper function that gets called with all these params passed in, then `await initializeApp(...)` here; or a Bootstrap component as you suggested. Either way this is fine for now.

Reply:
True. Either way pulling it out then requires moving some redux things out of code-studio and I think this is fine as is for now